File: implementations/typescript/packages/recovery/test/recovery-types.test.js
import assert from "node:assert/strict";
import test from "node:test";
import { RECOVERY_EVENT_TYPES, isRecoveryEventType } from "@axisrobo/harmovela-recovery";

test("recovery event types includes all 7 registry entries", () => {
  const expected = [
    "interruption.requested",
    "interruption.acknowledged",
    "interruption.saved",
    "interruption.resumed",
    "interruption.cancelled",
    "compensation.requested",
    "compensation.completed"
  ];
  for (const type of expected) {
    assert.equal(RECOVERY_EVENT_TYPES.has(type), true, `missing: ${type}`);
  }
  assert.equal(RECOVERY_EVENT_TYPES.size, expected.length);
});

test("isRecoveryEventType positives", () => {
  assert.equal(isRecoveryEventType("interruption.requested"), true);
  assert.equal(isRecoveryEventType("interruption.acknowledged"), true);
  assert.equal(isRecoveryEventType("interruption.saved"), true);
  assert.equal(isRecoveryEventType("interruption.resumed"), true);
  assert.equal(isRecoveryEventType("interruption.cancelled"), true);
  assert.equal(isRecoveryEventType("compensation.requested"), true);
  assert.equal(isRecoveryEventType("compensation.completed"), true);
});

test("isRecoveryEventType negatives", () => {
  assert.equal(isRecoveryEventType("task.submitted"), false);
  assert.equal(isRecoveryEventType("session.opened"), false);
  assert.equal(isRecoveryEventType("delegation.requested"), false);
  assert.equal(isRecoveryEventType(""), false);
});

File: implementations/java/src/test/java/com/axisrobo/harmovela/recovery/RecoveryTypesTest.java
package com.axisrobo.harmovela.recovery;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class RecoveryTypesTest {

    @Test
    void includesAll7RegistryEntries() {
        assertEquals(7, RecoveryTypes.EVENT_TYPES.size());
        assertTrue(RecoveryTypes.EVENT_TYPES.contains("interruption.requested"));
        assertTrue(RecoveryTypes.EVENT_TYPES.contains("interruption.acknowledged"));
        assertTrue(RecoveryTypes.EVENT_TYPES.contains("interruption.saved"));
        assertTrue(RecoveryTypes.EVENT_TYPES.contains("interruption.resumed"));
        assertTrue(RecoveryTypes.EVENT_TYPES.contains("interruption.cancelled"));
        assertTrue(RecoveryTypes.EVENT_TYPES.contains("compensation.requested"));
        assertTrue(RecoveryTypes.EVENT_TYPES.contains("compensation.completed"));
    }

    @Test
    void isRecoveryEventTypePositives() {
        assertTrue(RecoveryTypes.isRecoveryEventType("interruption.requested"));
        assertTrue(RecoveryTypes.isRecoveryEventType("interruption.acknowledged"));
        assertTrue(RecoveryTypes.isRecoveryEventType("interruption.saved"));
        assertTrue(RecoveryTypes.isRecoveryEventType("interruption.resumed"));
        assertTrue(RecoveryTypes.isRecoveryEventType("interruption.cancelled"));
        assertTrue(RecoveryTypes.isRecoveryEventType("compensation.requested"));
        assertTrue(RecoveryTypes.isRecoveryEventType("compensation.completed"));
    }

    @Test
    void isRecoveryEventTypeNegatives() {
        assertFalse(RecoveryTypes.isRecoveryEventType("task.submitted"));
        assertFalse(RecoveryTypes.isRecoveryEventType("session.opened"));
        assertFalse(RecoveryTypes.isRecoveryEventType("delegation.requested"));
        assertFalse(RecoveryTypes.isRecoveryEventType(""));
    }

    @Test
    void isRecoveryEventTypeHandlesNull() {
        assertFalse(RecoveryTypes.isRecoveryEventType(null));
    }
}
